nilearn/image/resampling.py

```diff
@@ -445,6 +445,14 @@ def resample_img(img, target_affine=None, target_shape=None,
     shape = img.shape
     affine = img.affine
 
+    # If later on we want to impute sform using qform add this condition
+    # and img.get_qform(coded=True) == (None, 0):
+    sform, sform_code = img.get_sform(coded=True)
+    if not sform_code:
```

Comment:
I might be missing something here, but this is not what was recommended by @effigies in #3168, which was:

```python
sform, sform_code = img.get_sform(coded=True)
qform, qform_code = img.get_qform(coded=True)
if not sform_code and not qform_code:
    warn("qform and sform codes are zero; this file may be corrupted. Using fallback LAS affine.")
```

Any reason you don't check `q_form` ?